panther_gpiod/include/panther_gpiod/gpio_driver.hpp

```diff
@@ -0,0 +1,265 @@
+  void gpio_monitor_on();
+
+  /**
+   * @brief Disables asynchronous monitoring of GPIO pin events.
+   */
+  void gpio_monitor_off();
+
+private:
+  std::unique_ptr<gpiod::line_request> create_line_request(gpiod::chip & chip, const GPIOpin pin);
+  std::unique_ptr<gpiod::line_request> create_line_request(
+    gpiod::chip & chip, const std::vector<GPIOpin> & pins);
+  gpiod::line_settings generate_line_settings(const GPIOinfo & pin_info);
+  GPIOpin get_pin_from_offset(gpiod::line::offset offset) const;
+  GPIOinfo & get_pin_info_ref(GPIOpin pin);
+  void configure_line_request(gpiod::chip & chip, gpiod::request_builder & builder, GPIOpin pin);
+  void monitor_async_events();
+  void handle_edge_event(const gpiod::edge_event & event);
+  bool is_gpio_monitor_thread_running() const;
+
+  /**
+   * @brief Callback function for GPIO edge events.
+   *
+   * @param gpio_info Information related to the state of the GPIO pin for which the event took
+   * place.
+   */
+  std::function<void(const GPIOinfo & gpio_info)> gpio_edge_event_callback;
+
+  /**
+   * @brief Mapping of GPIO pins to their respective names.
+   *
+   * This map contains the names associated with different GPIO pins.
+   */
+  const std::map<GPIOpin, std::string> pin_names_{
+    {GPIOpin::WATCHDOG, "WATCHDOG"},
+    {GPIOpin::AUX_PW_EN, "AUX_PW_EN"},
+    {GPIOpin::CHRG_DISABLE, "CHRG_DISABLE"},
+    {GPIOpin::CHRG_SENSE, "CHRG_SENSE"},
+    {GPIOpin::DRIVER_EN, "DRIVER_EN"},
+    {GPIOpin::E_STOP_RESET, "E_STOP_RESET"},
+    {GPIOpin::FAN_SW, "FAN_SW"},
+    {GPIOpin::GPOUT1, "GPOUT1"},
+    {GPIOpin::GPOUT2, "GPOUT2"},
+    {GPIOpin::GPIN1, "GPIN1"},
+    {GPIOpin::GPIN2, "GPIN2"},
+    {GPIOpin::LED_SBC_SEL, "LED_SBC_SEL"},
+    {GPIOpin::SHDN_INIT, "SHDN_INIT"},
+    {GPIOpin::VDIG_OFF, "VDIG_OFF"},
+    {GPIOpin::VMOT_ON, "VMOT_ON"},
+  };
+
+  /**
+   * @brief Vector containing GPIO pin configuration information.
+   *
+   * This vector stores information related to GPIO pins such as pin configuration,
+   * direction, value, etc.
+   */
+  std::vector<GPIOinfo> gpio_info_;
+
+  /**
+   * @brief Mutex for managing access to GPIO pin information.
+   *
+   * This shared mutex allows controlling access to GPIO pin information to prevent race conditions.
+   */
+  mutable std::shared_mutex gpio_info_mutex_;
+
+  /**
+   * @brief Request object for controlling GPIO lines.
+   *
+   * This unique pointer manages the request for controlling GPIO lines.
+   */
+  std::unique_ptr<gpiod::line_request> line_request_;
+
+  /**
+   * @brief Thread object for monitoring GPIO events.
+   *
+   * This unique pointer manages the thread responsible for monitoring GPIO events asynchronously.
+   */
+  std::unique_ptr<std::thread> gpio_monitor_thread_;
+  std::atomic<bool> gpio_monitor_thread_enabled_{false};
+  static constexpr unsigned gpio_debounce_period_ = 10;
+  static constexpr unsigned edge_event_buffer_size_ = 2;
+  const std::filesystem::path gpio_chip_path_ = "/dev/gpiochip0";
```

Comment:
Shouldn't this be static constexpr too?
```suggestion
  static constexpr std::filesystem::path gpio_chip_path_ = "/dev/gpiochip0";
```

Reply:
```bash
a constexpr variable must have a literal type or a reference typeC/C++(2402)
```
I've tried it before :p